File: tournaments/stringsCrossover/stringsCrossover.cpp
int stringsCrossover(std::vector<std::string> inputArray,
                     std::string result) {

  int answer = 0;

  for (int i = 0; i < inputArray.size(); i++) {
    for (int j = i + 1; j < inputArray.size(); j++) {
      bool correct = true;
      for (int k = 0; k < result.size(); k++) {
        if (result[k] != inputArray[i][k]
            && result[k] != inputArray[j][k]) {
          correct = false;
          break;
        }
      }
      if (correct) {
        answer++;
      }
    }
  }
  return answer;
}

File: tournaments/stringsCrossover/stringsCrossover.js
function stringsCrossover(inputArray, result) {
    const n = inputArray.length;
    let r = 0;

    for (let i = 0; n > i; ++i) {
        for (let j = i + 1; n > j; ++j) {
            let a = 1;
            for (let k = 0; result.length > k; ++k) {
                if (result[k] !== inputArray[i][k] && result[k] !== inputArray[j][k]) {
                    a = 0;
                    break;
                }
            }
            if (a) ++r;
        }
    }

    return r;
}
